Add GSI monitor scripts

WalterKolczynski-NOAA opened the pull request:
**Description**
As part of the overarching goal of moving global workflow scripts out of component repositories and under global workflow, all of the scripts currently linked in GSI monitor are added to global workflow. Scripts have been updated to reflect standardization already made to other workflow scripts (preamble, no backticks, etc.). See [7aa637](https://github.com/NOAA-EMC/global-workflow/pull/969/commits/7aa637ca4c7856abd71889ac94d70ec84a504dcf) for those modifications specifically.

Related GSI-Montitor issue: NOAA-EMC/GSI-Monitor/pull/23

Fixes #967

**Type of change**

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
- [x] Cycled test on Hera
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

WalterKolczynski-NOAA committed:
Absorb GSI monitor scripts

Copies scripts out of the GSI monitor repo into global workflow.

Refs: #967

WalterKolczynski-NOAA committed:
Update GSI monitor scripts to use workflow conventions

Updates the new GSI monitor scripts to incorporate some of the improvements
already added to other workflow scripts:
- Let preamble handle script entry/exit messages and bash set
- Replace backticks with $( ) for subshells
- Remove env prints

Refs: #967

WalterKolczynski-NOAA commented:
Still waiting for another cycle or two to complete on Hera.

KateFriedman-NOAA commented on .gitignore:
```diff
@@ -178,19 +170,9 @@ ush/global_chgres.sh
-ush/minmon_xtrct_reduct.pl
 ush/mkwfsgbl.sh
-ush/ozn_xtrct.sh
 ush/radmon_ck_stdout.sh
```
Still ignoring `ush/radmon_ck_stdout.sh`? I also don't see this script coming over.

WalterKolczynski-NOAA replied:
It wasn't in link_workflow.sh, which theoretically means we don't use it, so I didn't bring it over. Probably means it also doesn't need to be ignored though.

KateFriedman-NOAA replied:
Okie dokie, yeah please also remove it from being ignored then. Thanks!

WalterKolczynski-NOAA commented:
Added a companion issue to GSI-Monitor that removes the scripts added here.

WalterKolczynski-NOAA commented:
So, after running this I realize we don't test these jobs in any of our current development layouts. @EdwardSafford-NOAA @aerorahul can this be tested (or at least mostly tested) otherwise, or just fix it later once we have it set up to run?

EdwardSafford-NOAA commented:
@WalterKolczynski-NOAA I can run the test driver scripts.  It's not a robust test but it will be something.  I'll try to make that happen this morning and will advise.

aerorahul commented:
From `config.vrfy`:

![Screen Shot 2022-08-15 at 9 44 37 AM](https://user-images.githubusercontent.com/11394126/184647029-369b65cf-a0fb-46c9-8924-a228659a5be9.png)

Do these not trigger the GSI monitoring jobs/scripts?

EdwardSafford-NOAA commented:
Unfortunately the test-driver scripts will require some work to run with these changes.  It won't be difficult but it's more than I can do quickly today.  

I don't know a great deal (yet) about the global-workflow, but from what I see @aerorahul is right -- the settings in config.vrfy should set the 3 monitor jobs to run.

WalterKolczynski-NOAA commented:
> From `config.vrfy`:
> 
> ![Screen Shot 2022-08-15 at 9 44 37 AM](https://user-images.githubusercontent.com/11394126/184647029-369b65cf-a0fb-46c9-8924-a228659a5be9.png)
> 
> Do these not trigger the GSI monitoring jobs/scripts?

Oh, they might. gempak and awips get turned on in config.base, so I didn't think to look anywhere else. Will check.

WalterKolczynski-NOAA commented:
> > From `config.vrfy`:
> > ![Screen Shot 2022-08-15 at 9 44 37 AM](https://user-images.githubusercontent.com/11394126/184647029-369b65cf-a0fb-46c9-8924-a228659a5be9.png)
> > Do these not trigger the GSI monitoring jobs/scripts?
> 
> Oh, they might. gempak and awips get turned on in config.base, so I didn't think to look anywhere else. Will check.

They don't appear to. I actually already had them on for my run and I don't see any monitoring jobs. Unless they are run as part of another job name (they shouldn't, since they have j-jobs).

WalterKolczynski-NOAA committed:
Remove unused script from .gitignore

`ush/radmon_ck_stdout.sh` was being ignored, but it was not even being
linked by the link script anymore, so the ignore is unnecessary.

Refs: #967

RussTreadon-NOAA commented:
`config.vrfy` defines `TANKverf` variables under the `VRFYRAD`, `VRFYMINMON`, and `VRFYOZN` sections.   I believe monitoring output is written to the `TANKverf` directories.  The rocoto workflow executes the monitoring jobs from `vrfy.sh`.    This differs from operations.  A check of `gdasvrfy.log` will show us if (a) the monitoring jobs were run and (b) where the output is saved.   

We (developers) run some jobs which operations does not run.   For other jobs which both ops and dev run, we sometimes run the jobs from difference places in the workflow and write results to different directory structures.

aerorahul commented:
> `config.vrfy` defines `TANKverf` variables under the `VRFYRAD`, `VRFYMINMON`, and `VRFYOZN` sections. I believe monitoring output is written to the `TANKverf` directories. The rocoto workflow executes the monitoring jobs from `vrfy.sh`. This differs from operations. A check of `gdasvrfy.log` will show us if (a) the monitoring jobs were run and (b) where the output is saved.
> 
> We (developers) run some jobs which operations does not run. For other jobs which both ops and dev run, we sometimes run the jobs from difference places in the workflow and write results to different directory structures.

Thanks @RussTreadon-NOAA. Your feedback and in-depth knowledge of the system is greatly appreciated.
This part of the workflow is not exercised by all developers and so it remains somewhat unknown to most who do not exercise data assimilation components.  
We will hope to do this better, starting with perhaps not including a number of activities in the same `vrfy` job.  A lot gets swept under this one job.

@WalterKolczynski-NOAA 
When we run the next "cycled" test, see if we can run `vrfy` and capture some of the DA verification and monitoring jobs output.  We might need to run for a few cycles.
If you are running a test with this branch, I can take a look.  I think I remember enough to tell the difference if this is being exercised or not.

WalterKolczynski-NOAA commented:
Okay, at least some of these scripts *are* running. Currently `exgdas_atmos_verfrad.sh` is dying on a failed move for abi_g16. I can add checks that the gunzip file exists before trying to rename and expand it, but are there any SATYPEs that should cause a fail if missing?

EdwardSafford-NOAA commented:
Yikes!  Missing diag files are common and shouldn't cause the script to die, but rather be reported as missing in the warning report.  Can you point me to your log file?  I'd like to take a look.

bubba0077 commented:
> Yikes! Missing diag files are common and shouldn't cause the script to die, but rather be reported as missing in the warning report. Can you point me to your log file? I'd like to take a look.

This is just a side-effect of having `set -e` on now. I just wanted to confirm none of these should cause a failure.

WalterKolczynski-NOAA commented:
> Yikes! Missing diag files are common and shouldn't cause the script to die, but rather be reported as missing in the warning report. Can you point me to your log file? I'd like to take a look.

This is just a side-effect of having set -e on now. I just wanted to confirm none of these should cause a failure.

EdwardSafford-NOAA reviewed (approved):
Looks ok to me.

WalterKolczynski-NOAA committed:
Add missing radmon script

One radmon script is needed but was not previously linked, so it hadn't
been added to the repository.

Refs: #967

WalterKolczynski-NOAA committed:
Fixes GSI monitor undefined and error exits

After moving the GSI monitor scripts to workflow, several of the scripts
would fail because set -eu is on, either due to undefined variables or
trying to operate on non-existent files. This corrects some of those issues.

Some of the non-existent files were globs, for these the bash built-in
`compgen -G` is used to check if it resolves to any files.

Refs: #967

WalterKolczynski-NOAA commented:
Finally worked through all the issues due to undefined variables or operating on missing files.

@EdwardSafford-NOAA Please carefully review specifically those changes in the most recent commit (cb29351) to make sure I haven't caused any unwanted behavior with my changes.

EdwardSafford-NOAA commented on ush/radmon_verf_angle.sh:
```diff
@@ -0,0 +1,251 @@
+             fail=$(expr $fail + 1)
+         fi
+
+#-------------------------------------------------------------------
```
This is on me -- the comments from 195-209 are out of date and can be removed.

WalterKolczynski-NOAA replied:
Removed

EdwardSafford-NOAA commented on scripts/exgdas_atmos_verfrad.sh:
```diff
@@ -0,0 +1,225 @@
+export PDATE=${PDY}${cyc}
+export DO_DIAG_RPT=${DO_DIAG_RPT:-1}
+export DO_DATA_RPT=${DO_DATA_RPT:-1}
+export USE_MAIL=${USE_MAIL:-0}
```
The capability to mail warning reports directly from the data extraction was removed a while ago so the USE_MAIL, MAIL_TO, and MAIL_CC variables can be removed.

WalterKolczynski-NOAA replied:
Removed

EdwardSafford-NOAA commented on ush/radmon_verf_time.sh:
```diff
@@ -0,0 +1,574 @@
+MAIL_CC=${MAIL_CC:-}
+VERBOSE=${VERBOSE:-NO}
+LITTLE_ENDIAN=${LITTLE_ENDIAN:-0}
+USE_MAIL=${USE_MAIL:-1}
```
USE_MAIL, MAIL_TO, and MAIL_CC can be removed.

WalterKolczynski-NOAA replied:
Removed

EdwardSafford-NOAA reviewed (approved):
This looks good and thank you for making many improvements.  I noted some minor things that can be removed, either now or as a later step.

WalterKolczynski-NOAA committed:
Remove unused mail options from radmon

Removes unused mail options from radmon scripts. Also an outdated comment.

Refs: #967
